signals/webhook/webhook.go

```diff
@@ -38,27 +37,67 @@ const (
 
 // Note: micro requires stateless operation so the Listen() method should not use the
 // receive struct to save or modify state.
-type webhook struct{}
+// webhooks are special in that they should only have one http Server since the port is fixed at runtime
+// this means that webhook signals are stateful, however losing this state is not a concern since
+// the connections will be re-initialized by the SignalClient
+type webhook struct {
+	srv *http.Server
+	sync.RWMutex
+	inactiveEndpoints map[string]struct{}
+}
+
+// New creates a new webhook listener for the specified port
+func New(port int) sdk.Listener {
+	srv := &http.Server{
+		Addr: fmt.Sprintf(":%v", port),
+		// Good practice to enforce timeouts to avoid Slowloris attacks
+		WriteTimeout: time.Second * 5,
+		ReadTimeout:  time.Second * 5,
+		IdleTimeout:  time.Second * 30,
+	}
+	// Start http server
+	go func() {
+		log.Printf("starting http server listening on: %s", srv.Addr)
+		err := srv.ListenAndServe()
+		if err == http.ErrServerClosed {
```

Comment:
should all errors be panicked since the server should never be shutdown under normal operations?

Reply:
One situation where it might be needed to shutdown the server would be as part of restarting it. E.g. if some config option has been changed and we need to restart the server. Not sure if we do that today...

Reply:
in that case, someone can just delete the pod(s) for this signal, however I don't foresee a need to do that today